vaadin-spring/src/main/java/com/vaadin/flow/spring/security/VaadinAwareSecurityContextHolderStrategy.java

```diff
@@ -0,0 +1,69 @@
+package com.vaadin.flow.spring.security;
+
+import com.vaadin.flow.server.VaadinSession;
+import org.springframework.lang.NonNull;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolderStrategy;
+import org.springframework.security.core.context.SecurityContextImpl;
+import org.springframework.security.web.context.HttpSessionSecurityContextRepository;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public final class VaadinAwareSecurityContextHolderStrategy
```

Comment:
Javadoc is needed. Although this is internally used class, a couple of sentences describing that this implementation looks for the `VaadinSession` for security context and it provides the security context for `UI.access()` calls.

Reply:
Added javadoc